src/arm/Gos-linux.c

```diff
@@ -118,6 +118,7 @@ arm_handle_signal_frame (unw_cursor_t *cursor)
   dwarf_get (&c->dwarf, c->dwarf.loc[UNW_ARM_R15], &c->dwarf.ip);
 
   c->dwarf.pi_valid = 0;
+  c->dwarf.use_prev_instr = 0;
```

Comment:
so far the init seems to be done in the platform-specific init.h header. either this code here is duplicating that, or the other code can now be removed?

Reply:
Hi @milianw , thanks for reply. 
> so far the init seems to be done in the platform-specific init.h header

Can you point me the code? I only see the api support pass the `use_prev_instr` as a parameter.

I think this's necessary if the library can handle this case, there's no strong easy to let user to handle this. Btw, I saw the `aarch64` has done this in [`handle_signal_frame`](https://github.com/libunwind/libunwind/blob/55ef85c50e4184c98c03e0c52ffb1c39b5201fab/src/aarch64/Gstep.c#L128)

Reply:
see e.g. `src/arm/init.h:70` or `src/x86_64/init.h:81` etc.

Reply:
@milianw thanks for pointing the code. This initialization makes me a little bit confusing.

The local unwind need caller to tell libunwind whether the frame is signal frame via passing `UNW_INIT_SIGNAL_FRAME` as extend parameter. My question is: if we can determine the frame via `unw_is_signal_frame(1)`, why we don't hide this detail for users?

Reply:
I dunno, I have no clue about this code. I'm just saying that we don't need to init `use_prev_instr` like done in your patch, since it seems to be done already elsewhere?

Reply:
Thx for reply. No, not really. For remote mode, it's true. Because it never run into signal handler. But for local mode, it just let user to pass a flag to determine the value of f`use_prev_instr `

https://github.com/libunwind/libunwind/blob/d32956507cf29d9b1a98a8bce53c78623908f4fe/src/arm/Ginit_local.c#L62-L71

Reply:
Yea I agree this looks unnecessary.  Is there some argument for keeping it?